src/modules/menu/components/LayerDescriptionPopup.vue

```diff
@@ -38,47 +38,50 @@ const isExternal = computed(() => layer.value?.isExternal ?? false)
 const legends = computed(() => layer.value?.legends ?? [])
 
 watch(layer, async (newLayer) => {
-    htmlContent.value = await getLayerLegend(currentLang.value, newLayer.id)
+    htmlContent.value = await getLayerDescription(currentLang.value, newLayer.id)
 })
 
 watch(layerId, async (newLayerId) => {
-    htmlContent.value = await getLayerLegend(currentLang.value, newLayerId)
+    htmlContent.value = await getLayerDescription(currentLang.value, newLayerId)
 })
```

Comment:
Here we should also check for non null value of `newLayerId` and for non external layer like in `onMounted`. Although it is not directly related to this PR, but it would be wise to change it, to improve the robustness of the code.